storage/cost-analysis/storage_soft_delete_relative_cost_analyzer.py

```diff
@@ -0,0 +1,322 @@
+#!/usr/bin/env python
+
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Identifies buckets with relative increase in cost on enabling the soft-delete.
+
+The relative increase in cost of using soft delete is calculated by combining
+the storage/v2/deleted_bytes metric with the existing storage/v2/total_byte_seconds
+metric.
+
+Relative cost of each bucket = ('soft delete retention duration' 
+                                 × 'deleted bytes' / 'total bytes seconds' ) 
+                                 x 'cost of storing in storage class' 
+                                 x 'ratio of storage class'.
+"""
+
+import argparse
+import json
+import google.cloud.monitoring_v3 as monitoring_client
+
+# [START storage_soft_delete_relative_cost]
+
+
+def get_relative_cost(storage_class):
```

Comment:
We aren't yet requiring type hints, but probably will for new code soon, so go ahead and add them if you can. Search for Python type hints or PIP 484 for details.